api/src/test/java/org/openmrs/module/fgh/mpi/MpiContextTest.java

```diff
@@ -0,0 +1,135 @@
+	@Before
+	public void setup() {
+		PowerMockito.mockStatic(Context.class);
+		PowerMockito.mockStatic(MpiUtils.class);
+		PowerMockito.mockStatic(FhirUtils.class);
+		PowerMockito.mockStatic(MpiContext.class);
+		PowerMockito.mockStatic(KeyStore.class);
+		PowerMockito.mockStatic(KeyManagerFactory.class);
+		Whitebox.setInternalState(snapshotEventProcessor, MpiHttpClient.class, mockMpiHttpClient);
+		when(Context.getAdministrationService()).thenReturn(adminService);
+		when(adminService.getGlobalProperty(GP_AUTHENTICATION_TYPE)).thenReturn(AUTHENTICATION_TYPE.toString());
+		when(adminService.getGlobalProperty(GP_MPI_BASE_URL)).thenReturn(MPI_BASE_URL);
+		when(adminService.getGlobalProperty(GP_MPI_SYSTEM)).thenReturn(MPI_SYSTEM.toString());
+		when(adminService.getGlobalProperty(GP_MPI_APP_CONTENT_TYPE)).thenReturn(MPI_APP_CONTENT_TYPE);
+		when(adminService.getGlobalProperty(GP_SANTE_MESSAGE_HEADER_FOCUS_REFERENCE)).thenReturn(MESSAGE_HEADER_REFERENCE);
+		when(adminService.getGlobalProperty(GP_SANTE_MESSAGE_HEADER_EVENT_URI)).thenReturn(MESSAGE_HEADER_EVENT_URI);
+		when(adminService.getGlobalProperty(GP_SANTE_MESSAGE_HEADER_EVENT_URI)).thenReturn(MESSAGE_HEADER_EVENT_URI);
+		when(adminService.getGlobalProperty(GP_SANTE_CLIENT_ID)).thenReturn(SANTE_CLIENT_ID);
+		when(adminService.getGlobalProperty(GP_SANTE_CLIENT_SECRET)).thenReturn(SANTE_CLIENT_SECRET);
+		when(MpiUtils.getGlobalPropertyValue(GP_UUID_SYSTEM)).thenReturn(UUID_SYSTEM);
+	}
+	
+	@Test
+	public void init_shouldInitOauth() throws Exception {
+		AuthenticationType OUAUTH = AuthenticationType.OAUTH;
+		when(adminService.getGlobalProperty(GP_AUTHENTICATION_TYPE)).thenReturn(OUAUTH.toString());
+		when(adminService.getGlobalProperty(GP_MPI_SYSTEM)).thenReturn(MPI_SYSTEM.toString());
+		MpiContext initOauthContext = new MpiContext();
+		initOauthContext.init();
+		
+		assertEquals(OUAUTH, initOauthContext.getAuthenticationType());
+		assertEquals(MPI_BASE_URL, initOauthContext.getServerBaseUrl());
+		assertEquals(MPI_SYSTEM, initOauthContext.getMpiSystem());
+		assertEquals(UUID_SYSTEM, initOauthContext.getOpenmrsUuidSystem());
+		assertEquals(SANTE_CLIENT_ID, initOauthContext.getClientId());
+		assertEquals(SANTE_CLIENT_SECRET, initOauthContext.getClientSecret());
+		assertTrue(initOauthContext.isContextInitialized());
+	}
+	
+	@Test
+	public void init_shouldInitSSL() throws Exception {
+		KeyStore keyStoreMock = PowerMockito.mock(KeyStore.class);
+		KeyManagerFactory keyManagerFactoryMock = PowerMockito.mock(KeyManagerFactory.class);
+		SSLContext sslContextMock = PowerMockito.mock(SSLContext.class);
+		
+		AuthenticationType CERTIFICATE = AuthenticationType.CERTIFICATE;
+		when(adminService.getGlobalProperty(GP_AUTHENTICATION_TYPE)).thenReturn(CERTIFICATE.toString());
+		when(adminService.getGlobalProperty(GP_MPI_SYSTEM)).thenReturn(MPI_SYSTEM_AS_OPENCR.toString());
+		when(adminService.getGlobalProperty(GP_KEYSTORE_PATH)).thenReturn("src/test/resources/test/test-resource.xml");
```

Comment:
`test-resource.xml` is an empty file? Do you need it? If not, delete it from the project.